### ISSUE TITLE
[BUG] When token modes the hgidden token flashes visible on the map 

### ISSUE THREAD

**Issue by gioppoluca** [p4535992/conditional-visibility#9]
**Module Version:** v0.3.5

**Describe the bug**
NPC Token hidden, with roll higher than passive perception, PC moves nearby and the NPC token flashes visible.

**To Reproduce**
Hide teh token, move the other

**Expected behavior**
No flashing

**Screenshots**
Not possible since it is too fast to shot, but not enough not to see

**Browser:**
 - chrome

**Foundry Version:** 
v9

**Game System:** 
5e
**Additional context**

**Comment by lucasmiranda2711** [comment]
I actually found why this issue is happening, it has to do with  the document update trough the game emit, when it's updated it only considers one of the updated variables from the array because it's on the same id, to be more clear what's hapenning is:

```
if (effect.parent.isOwner) {
                    ConditionalVisibility.INSTANCE.actorUpdates.push({
                        _id: effect.parent.id,
                        [baseflag + status.visibilityId]: false,
                    });
                }
                if (Array.from(this.effectsByCondition().values()).filter((e) => effect.parent.getFlag(CONDITIONAL_VISIBILITY_MODULE_NAME, e.visibilityId) ?? false).length == 1) {
                    if (effect.parent.isOwner) {
                         ConditionalVisibility.INSTANCE.actorUpdates.push({                   //we shouldn't add another item to our array with the same id
                             _id: effect.parent.id,
                             [baseflag + 'hasEffect']: false,
                         });
                    }

```

Instead I'd suggest we do something like:

```
if (effect.parent.isOwner) {
                if (Array.from(this.effectsByCondition().values()).filter((e) => effect.parent.getFlag(CONDITIONAL_VISIBILITY_MODULE_NAME, e.visibilityId) ?? false).length == 1) {
                    if (effect.parent.isOwner) {
                       ConditionalVisibility.INSTANCE.actorUpdates.push({
                        _id: effect.parent.id,
                        [baseflag + status.visibilityId]: false,[baseflag + 'hasEffect']: false    //adding the two conditions on the same push
                    });
                    }
                    setTimeout(() => {
                        effect.parent.getActiveTokens()?.forEach((e) => {
                            e.alpha = 1;
                            e.visible = true;
                            e.data.hidden = false;
                        });
                    }, 350);
                }
else{
                    ConditionalVisibility.INSTANCE.actorUpdates.push({
                        _id: effect.parent.id,
                        [baseflag + status.visibilityId]: false,
                    });
                }}
```

**Comment by p4535992** [comment]
This issue is closed, because the new series 0.4.X is just to different, checkout the last version of the module.
